generic-sensor/generic-sensor-tests.js

```diff
@@ -141,11 +175,15 @@ function runGenericSensorTests(sensorType) {
     }
   }, "no exception is thrown when calling stop() on already stopped sensor");
 
-  async_test(t => {
-    window.onmessage = t.step_func(e => {
-      assert_equals(e.data, "SecurityError");
-      t.done();
-    });
+  promise_test(t => {
+    return load_iframe(make_script(sensorType))
+      .then(iframe => {
+        iframe.contentWindow.postMessage({}, '*');
```

Comment:
I was referring to a different problem, first message is posted to an iframe so that it can try constructing sensor object and reply back, then Promise starts listening to message. I'm not sure whether reply from the iframe can be lost.

> Also, this test fails with timeout error.

Do you have same issue with the test?